hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/AppThrottleAdviser.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.throttle;
+
+import static com.hedera.hapi.node.base.HederaFunctionality.CONTRACT_CALL;
+import static com.hedera.hapi.node.base.HederaFunctionality.CONTRACT_CREATE;
+import static com.hedera.hapi.node.base.ResponseCodeEnum.SUCCESS;
+import static com.hedera.hapi.util.HapiUtils.functionOf;
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.base.HederaFunctionality;
+import com.hedera.hapi.node.base.SignatureMap;
+import com.hedera.hapi.node.state.throttles.ThrottleUsageSnapshot;
+import com.hedera.hapi.util.UnknownHederaFunctionality;
+import com.hedera.node.app.spi.throttle.ThrottleAdviser;
+import com.hedera.node.app.workflows.TransactionInfo;
+import com.hedera.node.app.workflows.handle.record.RecordListBuilder;
+import com.hedera.node.app.workflows.handle.stack.SavepointStackImpl;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.time.Instant;
+import java.util.List;
+import java.util.Objects;
+import javax.inject.Inject;
+
+/**
+ * Default implementation of {@link ThrottleAdviser}.
+ */
+public class AppThrottleAdviser implements ThrottleAdviser {
+
+    private final NetworkUtilizationManager networkUtilizationManager;
+    private final Instant consensusNow;
+    private final RecordListBuilder recordListBuilder;
+    private final SavepointStackImpl stack;
+
+    @Inject
+    public AppThrottleAdviser(
+            @NonNull final NetworkUtilizationManager networkUtilizationManager,
+            @NonNull final Instant consensusNow,
```

Comment:
Some documented context on this timestamp would probably be helpful

Reply:
It contains the consensus time of the current transaction. We use it all over the place. Do you look for documentation that explains consensus time in general and how we use it?

Reply:
Yes, the definition of `consensusNow` is clear and ubiquitous. What isn't clear is the scope of `consensusNow`, especially in relation to the other parameters here. With all the dagger refactoring I'm quite confused about all the scopes, but perhaps I just need to get accustomed to it